apps/chat-e2e/src/tests/ChatSwitch.test.ts

```diff
@@ -0,0 +1,160 @@
+import { Conversation } from '@/chat/types/chat';
+import dialTest from '@/src/core/dialFixtures';
+import {
+  API,
+  ExpectedConstants,
+  ExpectedMessages,
+  MenuOptions,
+} from '@/src/testData';
+import { Cursors, Styles } from '@/src/ui/domData';
+import { responseThrottlingTimeout } from '@/src/ui/pages';
+import { expect } from '@playwright/test';
+
+dialTest(
+  'Another chat is not available while AI is generating a response.\n' +
+    'Chat menu is not available while AI is generating a response.\n' +
+    'Switching to another chat is not available while AI is replaying a chat\n' +
+    'Switching to another chat is not available while AI is regenerating response in compare mode',
+  async ({
+    dialHomePage,
+    conversations,
+    conversationData,
+    dataInjector,
+    setTestIds,
+    localStorageManager,
+    chat,
+    chatMessages,
+    compareConversationSelector,
+    compareConversation,
+  }) => {
+    setTestIds(
+      'EPMRTC-598',
+      'EPMRTC-599',
+      'EPMRTC-600',
+      'EPMRTC-601',
+      'EPMRTC-602',
+    );
+    const request =
+      'give me a sci-fi story with a main topic of your choice. 200 tokens minimum';
+    let firstConversation: Conversation;
+    let secondConversation: Conversation;
+    let thirdConversationName;
+
+    const verifyConversationCursor = async (
```

Comment:
maybe, it's better to move this method to `conversationAssertion.ts` ?

Reply:
done